extension/jdtls.ext/com.microsoft.gradle.bs.importer/src/com/microsoft/gradle/bs/importer/NamedPipeStream.java

```diff
@@ -55,35 +56,47 @@ public OutputStream getOutputStream() throws IOException {
         private void initializeNamedPipe() {
             String pathName = generateRandomPipeName();
             sendImporterPipeName(pathName);
-
             File pipeFile = new File(pathName);
 
-            // Need to retry until the pipeName was sent and pipe is created by Extension side
+            int attempts = 0;
             boolean connected = false;
-            while (!connected) {
-                try{
-                    if (isWindows()) {
-                        AsynchronousFileChannel channel = AsynchronousFileChannel.open(pipeFile.toPath(),
-                        StandardOpenOption.READ, StandardOpenOption.WRITE);
-                        input = new NamedPipeInputStream(channel);
-                        output = new NamedPipeOutputStream(channel);
-                    } else {
-                        UnixDomainSocketAddress socketAddress = UnixDomainSocketAddress.of(pipeFile.toPath());
-                        SocketChannel channel = SocketChannel.open(StandardProtocolFamily.UNIX);
-                        channel.connect(socketAddress);
-                        input = new NamedPipeInputStream(channel);
-                        output = new NamedPipeOutputStream(channel);
-                    }
+            // Need to retry until the pipeName was sent and pipe is created by Extension side
+            while (!connected && attempts < MAX_ATTEMPTS) {
+                try {
+                    attemptConnection(pipeFile);
                     connected = true;
                 } catch (IOException e) {
-                    try {
-                        Thread.sleep(1000);
-                    } catch (InterruptedException ie) {
-                        Thread.currentThread().interrupt();
-                        throw new RuntimeException("Thread interrupted while trying to connect to named pipe", ie);
-                    }
+                    handleConnectionFailure(e, attempts);
+                    attempts++;
                 }
             }
+            if (!connected) {
```

Comment:
`connected` and `attempts` are redundant. You can achieve the same purpose to check `attempts`